### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ikhoon**
- Caffeine 2.9.0 -> 2.9.1
- Dropwizard Metrics 4.1.20 -> 4.1.21
- java-jwt 3.15.0 -> 3.16.0
- Micrometer 1.6.6 -> 1.7.0
- Reactor 3.4.5 -> 3.4.6
- scala-collection-compat 2.4.3 -> 2.4.4
- scala-java8-compat 0.9.1 -> 1.0.0
- Spring 5.3.6 -> 5.3.7
- Build
  - Awaitibility 4.0.3 -> 4.1.0
  - Dagger 2.35 -> 2.35.1
  - Finagle 21.3.0 -> 21.4.0
  - gax-grpc 1.63.0 -> 1.64.0
  - gradle-node-plugin 3.0.1 -> 3.1.0
  - jmh-core 1.29 -> 1.31
  - Mockito 3.9.0 -> 3.10.0
  - Monix 3.3.0 -> 3.4.0
  - MUnit 0.7.25 -> 0.7.26
- Various front dependencies

### CODE REVIEW

**Review by trustin — APPROVED** [review]

File: core/src/test/java/com/linecorp/armeria/internal/common/metric/CaffeineMetricSupportTest.java [hunk 1]
```diff
@@ -250,7 +250,7 @@ void update(long hitCount, long missCount,
         void update(long hitCount, long missCount, long loadSuccessCount, long loadFailureCount,
                     long totalLoadTime, long evictionCount, long evictionWeight, long estimatedSize) {
             stats = new CacheStats(hitCount, missCount, loadSuccessCount, loadFailureCount,
-                                   totalLoadTime, evictionCount, evictionWeight);
+                                  totalLoadTime, evictionCount, evictionWeight);
```
Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/internal/common/metric/CaffeineMetricSupportTest.java:253]: revert?

**Review by minwoox — COMMENTED** [review]
Seems like the NPM build failed. 🤔

File: docs-client/package.json [hunk 2]
```diff
@@ -26,25 +26,25 @@
...
     "@types/material-ui": "^0.21.8",
     "@types/node": "^14.14.37",
-    "@types/react": "^17.0.3",
-    "@types/react-dom": "^17.0.3",
-    "@types/react-helmet": "^6.1.0",
+    "@types/react": "^17.0.5",
+    "@types/react-dom": "^17.0.5",
+    "@types/react-helmet": "^6.1.1",
     "@types/react-router-dom": "^5.1.7",
     "@types/react-select": "^4.0.14",
```
Comment (KarboniteKream) [inline docs-client/package.json:42]: We can actually remove this. I removed the actual library in #3167, but forgot to remove the types 😅

Comment (ikhoon) [reply docs-client/package.json:42]: Oh, nice catch!!

**Comment by ikhoon** [comment]
> Seems like the NPM build failed. 🤔

There are major version updates that make breaking changes. 😭
I reverted them. Let me deal with the changes in a separate PR.

**Pull request merged**